tests/unittests/sources/test_smartos.py

```diff
@@ -1412,40 +1481,15 @@ class TestSerialConcurrency(CiTestCase):
     This takes on the order of 2 to 3 minutes to run.
     """
 
-    allowed_subp = ["mdata-get"]
-
-    def setUp(self):
-        self.mdata_proc = multiprocessing.Process(target=self.start_mdata_loop)
-        self.mdata_proc.start()
-        super(TestSerialConcurrency, self).setUp()
-
-    def tearDown(self):
-        # os.kill() rather than mdata_proc.terminate() to avoid console spam.
-        os.kill(self.mdata_proc.pid, signal.SIGKILL)
-        self.mdata_proc.join()
-        super().tearDown()
-
-    def start_mdata_loop(self):
-        """
-        The mdata-get command is repeatedly run in a separate process so
-        that it may try to race with metadata operations performed in the
-        main test process.  Use of mdata-get is better than two processes
-        using the protocol implementation in DataSourceSmartOS because we
-        are testing to be sure that cloud-init and mdata-get respect each
-        others locks.
-        """
-        rcs = list(range(256))
-        while True:
-            subp(["mdata-get", "sdc:routes"], rcs=rcs)
-
-    def test_all_keys(self):
-        self.assertIsNotNone(self.mdata_proc.pid)
+    @pytest.mark.allow_subp_for("mdata-get")
```

Comment:
And TIL about joyent's mdata-get command. Glad this test is skipIf'd  under `get_smartos_environ() != SMARTOS_ENV_KVM`